be/test/exprs/vectorized/array_functions_test.cpp

```diff
@@ -2906,6 +2932,911 @@ TEST_F(ArrayFunctionsTest, array_reverse_only_null) {
+    src_column->append_datum(DatumArray{(int16_t)2, (int16_t)3, (int16_t)7, (int16_t)8});
+    src_column->append_datum(DatumArray{(int16_t)4, (int16_t)3, (int16_t)2, (int16_t)1});
+    src_column->append_datum(Datum());
+    src_column->append_datum(DatumArray{(int16_t)4, Datum(), (int16_t)2, (int16_t)1});
+
+    auto offset_column = Int64Column::create();
+    offset_column->append(1);
+    offset_column->append(2);
+    offset_column->append(3);
+    offset_column->append(1);
+    offset_column->append(2);
+
+    auto length_column = Int64Column::create();
+    length_column->append(1);
+    length_column->append(3);
+    length_column->append(2);
+    length_column->append(1);
+    length_column->append(2);
+
+    ArraySlice<PrimitiveType::TYPE_SMALLINT> slice;
+    auto dest_column = slice.process(nullptr, {src_column, offset_column, length_column});
+
+    ASSERT_EQ(dest_column->size(), 5);
+    _check_array<int16_t>({(int16_t)5}, dest_column->get(0).get_array());
+    _check_array<int16_t>({(int16_t)3, (int16_t)7, (int16_t)8}, dest_column->get(1).get_array());
+    _check_array<int16_t>({(int16_t)2, (int16_t)1}, dest_column->get(2).get_array());
+    ASSERT_TRUE(dest_column->get(3).is_null());
+    ASSERT_TRUE(dest_column->get(4).get_array()[0].is_null());
+    ASSERT_EQ(2, dest_column->get(4).get_array()[1].get_int16());
+}
+
+TEST_F(ArrayFunctionsTest, array_slice_int) {
+    auto src_column = ColumnHelper::create_column(TYPE_ARRAY_INT, true);
+    src_column->append_datum(DatumArray{(int32_t)5, (int32_t)3, (int32_t)6});
+    src_column->append_datum(DatumArray{(int32_t)2, (int32_t)3, (int32_t)7, (int32_t)8});
+    src_column->append_datum(DatumArray{(int32_t)4, (int32_t)3, (int32_t)2, (int32_t)1});
+    src_column->append_datum(Datum());
+    src_column->append_datum(DatumArray{(int32_t)4, Datum(), (int32_t)2, (int32_t)1});
+
+    auto offset_column = Int64Column::create();
+    offset_column->append(1);
+    offset_column->append(2);
+    offset_column->append(3);
+    offset_column->append(1);
+    offset_column->append(2);
+
+    auto length_column = Int64Column::create();
+    length_column->append(1);
+    length_column->append(3);
+    length_column->append(2);
+    length_column->append(1);
+    length_column->append(2);
+
+    ArraySlice<PrimitiveType::TYPE_INT> slice;
+    auto dest_column = slice.process(nullptr, {src_column, offset_column, length_column});
+
+    ASSERT_EQ(dest_column->size(), 5);
+    _check_array<int32_t>({(int32_t)5}, dest_column->get(0).get_array());
+    _check_array<int32_t>({(int32_t)3, (int32_t)7, (int32_t)8}, dest_column->get(1).get_array());
+    _check_array<int32_t>({(int32_t)2, (int32_t)1}, dest_column->get(2).get_array());
+    ASSERT_TRUE(dest_column->get(3).is_null());
+    ASSERT_TRUE(dest_column->get(4).get_array()[0].is_null());
+    ASSERT_EQ(2, dest_column->get(4).get_array()[1].get_int32());
+}
+
+TEST_F(ArrayFunctionsTest, array_slice_bigint) {
```

Comment:
There's no significant differences between test cases for these numeric types, why duplicate them so many times?

Reply:
Because I think Its key focuse is to ensure that each type can be used by these functions. But duplicate cases is Annoying, so I avoid it in other functions.